### DISCUSSION TITLE
The Bot Leaves the server for no apparent reason.

### DISCUSSION THREAD

**Discussion by Midnighttreason** [PrismarineJS/mineflayer#1867] [Q&A] [answered]
The bot is currently supposed to start and connect to the server and then basically do a loop that calls the digDown function that so far only tells me that it ran. The problem is that the bot will join the server then only run the loop 4 times then the bot leaves the server but the code is still running in the console without any errors. So why does my bot leave instead of indefinitely calling the digDown function?
The code is hosted at this link.
https://pastebin.com/U1BrDB

Thank you for taking the time to read and help me understand this.

**Comment by amoraschi** [comment, marked as answer]
Hi, the code has expired, and for this type of question, it's faster if you join the Discord Server

**Reply by Midnighttreason** [reply]
I am somewhat new to JavaScript. Can you explain what “expired code” is or at least link an article that explains it? Also I will definitely join the discord.

**Reply by amoraschi** [reply]
Pastebin doesn't work, says it's expired

**Reply by Midnighttreason** [reply]
const mineflayer = require('mineflayer')
const vec3 = require('vec3')
const options = {
  host: 'localhost',
  port: parseInt(process.argv[2]),
  username: 'Bot',
}
const bot = mineflayer.createBot(options)
 
bot.once('login', ()=>{
    mcData = require('minecraft-data')(bot.version);
 
    mainLoop();
});
 
function mainLoop() {
  bot.chat("Staring mainLoop")
  digDown("north")
  setTimeout(mainLoop, 100);
}
 
function digDown(direction) {
  bot.chat("digDown function has been called.")
  switch (direction) {
    case "north":
    bot.chat("Case north called")
    break
}}

**Reply by Midnighttreason** [reply]
Also I am having trouble finding the discord server you mentioned earlier.

**Reply by amoraschi** [reply]
Probably because you are trying to run digDown every 100ms. Anyways at the top of the Readme there's a badge, and if you click it it will invite you

**Reply by Midnighttreason** [reply]
Thank you very much adjusting the time to a delay of 3 seconds fixed the problem. I found that any time less then that and the bot would still leave the game.
